ViAn/GUI/mainwindow.cpp

```diff
@@ -947,6 +1104,52 @@ void MainWindow::remove_item_from_tree(MyQTreeWidgetItem *my_item) {
 }
 
 /**
+ * @brief MainWindow::remove_analysis_from_queue
+ * @param my_item to be removed
+ * Removes the item from the queue.
+ */
+void MainWindow::remove_analysis_from_queue(MyQTreeWidgetItem *my_item) {
+    QQueue<MyQTreeWidgetItem*> *tmp_queue = new QQueue<MyQTreeWidgetItem*>();
+    int id_count = 1; // Used to keep track of where in the queue it is
+    while (!analysis_queue->isEmpty()){
```

Comment:
If I'm understanding this correctly, what you want to do is remove something in a QQueue with the pointer to it.
I feel you could use the function `indexOf(const T &)` and then `removeAt(int)` (see [QQueue Doc](http://doc.qt.io/qt-5/qqueue-members.html)).

*The same probably goes for `MainWindow::remove_analysis_of_video`.*